misc/changelog-generator/changelog-generator

```diff
@@ -20,13 +20,16 @@ LINKED_SHAS = {}
 # messages.
 SHA_TO_TRACKER = {}
 
+# more relaxed regexp to find JIRA issues anywhere in commit message
 JIRA_REGEX = r"(?:Jira:? *)?(?:https?://tracker.mender.io/browse/)?((?:CFE|ENT|INF|ARCHIVE|MEN|QA)-[0-9]+)"
+# more strict regexp to find JIRA issues only in the beginning of title
+JIRA_TITLE_REGEX = r"(?:CFE|ENT|INF|ARCHIVE|MEN|QA)-[0-9]+"
```

Comment:
Shouldn't this regex start with `^` then?

Reply:
not necessary. Later it's used in `re.match` function, which does an "ancored match" - i.e. implies `^` at the beginning of regexp. For non-anchored match, Python provides a `re.search` function:
```
lex@flower:~$ python3
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> re.match('a','abba')
<_sre.SRE_Match object; span=(0, 1), match='a'>
>>> re.match('b','abba')
>>> re.match('c','abba')
>>> re.search('b','abba')
<_sre.SRE_Match object; span=(1, 2), match='b'>
>>> re.search('c','abba')
>>> 
lex@flower:~$ 
```

but maybe you're right and it's better to include it for readability

Reply:
Yeah, the code is quite cryptic. I think it's better to be explicit.